src/vs/editor/contrib/suggest/suggestWidget.ts

```diff
@@ -1024,6 +1034,16 @@ export class SuggestWidget implements IContentWidget, IListVirtualDelegate<IComp
 		const widgetX = widgetCoords.left;
 		const widgetY = widgetCoords.top;
 
+		// Fixes #27649
+		// Check if the Y changed to the top of the cursor and keep the widget flagged to prefer top
+		if (this.docsPositionPreviousWidgetY &&
+			this.docsPositionPreviousWidgetY < widgetY &&
+			!this.preferDocPositionTop) {
+			this.preferDocPositionTop = true;
+			this.adjustDocsPosition();
```

Comment:
Why do we need to call `this.adjustDocsPosition` again here? None of the constants declared above this `if` condition gets changed by `this.preferDocPositionTop = true`

Reply:
I will double check tomorrow when I have a moment, but if I remember correctly I believe it's because at this point it has now flipped back to the bottom so I am forcing it to rerender back to the top. I will verify though.

Reply:
In that case, why not exit after calling `adjustDocsPosition`?

ie. Replace the `this.adjustDocsPosition()` here with
```
this.docsPositionPreviousWidgetY = widgetY;
this.adjustDocsPosition();
return;
```